govuk-one-login/authentication-frontend
# Origin/aut 2602/2hrs fix incorrect auth app incorrect codes behaviour

## What
When a user restarts the journey with new session variables, checking the session lockout is not enough, we have to check for an existing user lockout code from the backend api. 

## How to review

1. Code Review
2. Ensure environment variables 'SUPPORT_2HR_LOCKOUT' and 'SUPPORT_2FA_B4_PASSWORD_RESET' are set to 1
3. Deploy to sandpit with `./deploy-sandpit.sh -l` (or run local)
4. Proceed through PW reset journey and enter the correct OTP email
5. Enter 5 wrong AUTH_APP OTPS, check that you are redirected to the screen corresponding to the first Screenshot below
6. Restart the journey by going to localhost:2000, or going through the stub again, go to the reset-password journey and verify that you're redirected to the 2nd screenshot below, after entering the correct email OTP. 
7. Proceed through the SIGN_IN journey
8. Enter 5 wrong AUTH_APP OTPS, check that you are redirected to the screen corresponding to the first Screenshot below
9. Restart the journey by going to localhost:2000, or going through the stub again, go through the sign-in journey again
10. Verify that you're redirected to the 2nd screenshot below, after entering the correct password.


Expected 2hr screens to be redirected to during the PW reset and sign in journey, when entering too many wrong codes:


![image](https://github.com/govuk-one-login/authentication-frontend/assets/146068663/72e8f1d4-abad-42d4-acbb-2304d0da3b39)

Expected 2hr screens to be redirected to during PW reset, when user is locked and tries to complete the journey: 
![image](https://github.com/govuk-one-login/authentication-frontend/assets/146068663/25f621b4-8ee7-4db1-abb3-1eeea9772192)


## Change have been demonstrated

Changes to the user interface or content should be demonstrated to Content Design and Interaction Design before being merged. This is to ensure they can make any necessary changes to Figma.

The changes were demonstrated in a demo to Kim Clayden
